### PR TITLE
RecommenderXBlock: Drop py3.8 support | Replace pkg_resources lib with importlib.resources

### PR DESCRIPTION
**Pull request by beingfaisal**
Issue - [Move on from deprecated pkg_resources api](https://github.com/openedx/RecommenderXBlock/issues/96)
Issue - [Drop Python 3.8 & Add Support for Python 3.12](https://github.com/openedx/RecommenderXBlock/issues/100)

- Drop python 3.8 support and add support for python 3.12
- Replace pkg_resources with importlib

Screenshots
Screenshot 2024-09-10 at 1 05 41 PM
Screenshot 2024-09-10 at 1 06 17 PM

### CODE REVIEW

**Comment by mphilbrick211** [comment]
Hi @beingfaisal! Thanks for this contribution! If you are contributing on behalf of edly, please have your manager reach out to oscm@axim.org to add you to our existing agreement. Thanks!

**Comment by farhan** [comment]
@beingfaisal All seems ok to me, but will be great to cover following issue within this PR and add support of Python 3.12
https://github.com/openedx/RecommenderXBlock/issues/100

**Review by farhan — CHANGES_REQUESTED** [review]
Requrest an addition

**Comment by e0d** [comment]
@beingfaisal are you on the Arbisoft team?

**Comment by e0d** [comment]
@beingfaisal looks like your branch needs some conflicts resolved, can you have a look?  Otherwise the checks are all passing.

**Review by farhan — CHANGES_REQUESTED** [review]
1 change

File: recommender/__init__.py [hunk 1]
```diff
@@ -6,4 +6,4 @@
 # which is not loaded when running `manage.py` commands (which is used by `make compile_translations`)
 # from .recommender import RecommenderXBlock
 
-__version__ = '2.2.1'
+__version__ = '2.2.2'
```
Comment (farhan) [inline recommender/__init__.py:9]: Please update it to `3.0.0` as we are dropping support of Python 3.8

**Review by farhan — APPROVED** [review]

**Pull request merged**